crates/flox/src/commands/environment.rs

```diff
@@ -30,15 +33,71 @@ pub struct EnvironmentArgs {
 
 pub type EnvironmentRef = String;
 
+#[derive(Bpaf, Clone)]
+pub enum EnvironmentSelect {
+    Dir(
+        /// Path containing a .flox/ directory
+        #[bpaf(long("dir"), short('d'), argument("path"))]
+        PathBuf,
+    ),
+    Remote(
+        /// A remote environment on floxhub
+        #[bpaf(long("remote"), short('r'), argument("owner/name"))]
+        environment_ref::EnvironmentRef,
+    ),
+}
+
+impl Default for EnvironmentSelect {
+    fn default() -> Self {
+        EnvironmentSelect::Dir(PathBuf::from("./"))
+    }
+}
+
+impl EnvironmentSelect {
+    fn resolve(&self, temp_dir: impl AsRef<Path>) -> Result<ConcreteEnvironment> {
+        let env = match self {
+            EnvironmentSelect::Dir(path) => ConcreteEnvironment::Path(
+                PathEnvironment::open(path, "default".parse().unwrap(), temp_dir)
+                    .context("Couldn't open path environment")?,
+            ),
+
+            EnvironmentSelect::Remote(_) => todo!(),
+        };
+
+        Ok(env)
+    }
+}
+
+enum ConcreteEnvironment {
```

Comment:
I think we'll need `ConcreteEnvironment` in flox_rust_sdk and I'm using this commit for #280: https://github.com/flox/flox/commit/ecd50a7c4f8b81f7e7393c187c41ca157fe3f4d2

Reply:
why would we need that there (also the general `open` function)?
anyway should be no harm to move it.

Reply:
I'm thinking that should be the return type for this function, does that seem right?
https://github.com/flox/flox/pull/280/files#diff-427c173f56c02f5bc2b7b7cfae183460d3c75e1c1fa73398e2054dd01ff3d37cR348

Reply:
I'm not sure if that method really belongs there in that form.
you'd think it can open any impl of environments but its decidedly only for the things in .flox dirs.

To me that concrete thing is only relevant at the cli side.

We could add some `{Path,Managed}Environment::probe(path)` method that checks if the path contains either of those (they really should be mutually exclusive) and use that in the resolve (or whatever we call it) method when dealing with the arguments.

Reply:
We can revisit in my PR https://github.com/flox/flox/pull/280/files#r1349243216